lib/screenplay/alerts/cache.ex

```diff
@@ -0,0 +1,80 @@
+defmodule Screenplay.Alerts.Cache do
+  @moduledoc """
+  Module used to fetch current alerts from the V3 API every 4 seconds. Each set of alerts is saved to an internal ETS table.
+  """
+
+  require Logger
+
+  use GenServer
+
+  alias Screenplay.Alerts.Alert
+  alias Screenplay.V3Api
+
+  @default_opts [
+    get_json_fn: &V3Api.get_json/2,
+    update_interval_ms: 4_000
+  ]
+
+  # Client
+  def start_link(opts) do
+    opts = Keyword.merge(@default_opts, opts)
+    GenServer.start_link(__MODULE__, opts)
+  end
+
+  @doc """
+  Retrieves an alert struct given an alert ID.
+  """
+  @spec alert(String.t()) :: Alert.t()
+  def alert(alert_id) do
+    case :ets.match(:alerts, {alert_id, :"$1"}) do
+      [[alert]] -> alert
+      [] -> nil
+    end
+  end
+
+  @doc """
+  Retrieves the full set of alerts.
+  """
+  @spec alerts :: list(Alert.t())
+  def alerts do
+    :ets.select(:alerts, [{{:_, :"$1"}, [], [:"$1"]}])
+  end
+
+  # Server
+  @impl true
+  def init(opts) do
+    _ = :ets.new(:alerts, [:protected, :named_table, read_concurrency: true])
```

Comment:
Hmm, I thought leaving that out would cause a dialyzer issue, but I'm wrong about that. I've seen errors from dialyzer before when you don't use the return of a function (was `Logger` related stuff so I shouldn't have assumed this was the same). Will remove it.